src/sentry/incidents/tasks.py

```diff
@@ -140,3 +142,41 @@ def handle_trigger_action(action_id, incident_id, project_id, method):
         )
     )
     getattr(action, method)(incident, project)
+
+
+@instrumented_task(
+    name="sentry.incidents.tasks.auto_resolve_snapshot_incidents",
+    queue="incidents",
+    default_retry_delay=60 * 5,
+    max_retries=1,
+)
+@retry(exclude=(DeleteAborted,))
+def auto_resolve_snapshot_incidents(alert_rule_id, **kwargs):
+    from sentry.incidents.models import AlertRule
+    from sentry.incidents.logic import update_incident_status
+
+    try:
+        alert_rule = AlertRule.objects_with_snapshots.get(id=alert_rule_id)
+    except AlertRule.DoesNotExist:
+        return
+
+    if alert_rule.status != AlertRuleStatus.SNAPSHOT.value:
+        raise DeleteAborted
+
+    incidents = Incident.objects.filter(alert_rule=alert_rule).exclude(
+        status=IncidentStatus.CLOSED.value
+    )
+
+    if incidents:
+        has_more = incidents.count() > 1
+        incident = incidents.first()
+        update_incident_status(
+            incident,
+            IncidentStatus.CLOSED,
+            comment="This alert has been auto-resolved because the rule that triggered it has been modified or deleted.",
+        )
+
+    if has_more:
+        auto_resolve_snapshot_incidents.apply_async(
+            kwargs={"alert_rule_id": alert_rule_id}, countdown=1
+        )
```

Comment:
I think we probably want to batch these a little more. Maybe process 10-50 at a time, then fire off another task? Best way to handle this is to do `[:<batch_size> + 1]` on the queryset, then if the length of the results is `> batch_size` then `has_more` is true. Then you only process the first `batch_size` items and leave the extra one off.